hazelcast-spring/src/main/java/com/hazelcast/spring/AbstractHazelcastBeanDefinitionParser.java

```diff
@@ -514,6 +521,68 @@ protected BeanDefinition getPreloaderConfig(Node node) {
             return nearCachePreloaderConfigBuilder.getBeanDefinition();
         }
 
+        protected void handleNativeMemory(Node node) {
+            BeanDefinitionBuilder nativeMemoryConfigBuilder = createBeanBuilder(NativeMemoryConfig.class);
+            AbstractBeanDefinition beanDefinition = nativeMemoryConfigBuilder.getBeanDefinition();
+            fillAttributeValues(node, nativeMemoryConfigBuilder, "persistentMemoryDirectory");
+            ManagedList<BeanDefinition> directories = new ManagedList<>();
+            BeanDefinitionBuilder pmemConfigBuilder = createBeanBuilder(PersistentMemoryConfig.class);
+            for (Node child : childElements(node)) {
+                String nodeName = cleanNodeName(child);
+                if ("size".equals(nodeName)) {
+                    handleMemorySizeConfig(child, nativeMemoryConfigBuilder);
+                } else if ("persistent-memory".equals(nodeName)) {
+                    handlePersistentMemoryConfig(child, directories);
+                }
+            }
+
+            Node attrPmemDirectory = node.getAttributes().getNamedItem("persistent-memory-directory");
+            if (attrPmemDirectory != null) {
+                BeanDefinitionBuilder pmemDirConfigBuilder = createBeanBuilder(PersistentMemoryDirectoryConfig.class);
+                pmemDirConfigBuilder.addConstructorArgValue(getTextContent(attrPmemDirectory));
```

Comment:
Are we missing the numa node id parsing here?

Reply:
This parses the old-style `persistent-memory-directory` element that had no `numa-node` attribute. We have it for `persistent-memory/directories/directory` only that is parsed in `handlePersistentMemoryConfig`.